### ISSUE TITLE
RIDE 1.1 on Win 7 x86 gives TestRunnerAgent errors

### ISSUE THREAD

**Issue by yanne** [robotframework/RIDE#1257]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1218) by kaj... on 7 Mar 2013*

I installed RIDE 1.1 today and tried to execute Jybot test on it using Windows 7 x86 platform and got following errors while doing so:

Test Execution Errors

20130307 12:13:18.842 	ERROR 	Calling listener method 'start_suite' of listener 'C:\Python27\lib\site-packages\robotide\contrib\testrunner\TestRunnerAgent.py' failed: error: (55, 'Software caused connection abort')
20130307 12:13:18.858 	ERROR 	Calling listener method 'start_test' of listener 'C:\Python27\lib\site-packages\robotide\contrib\testrunner\TestRunnerAgent.py' failed: error: (55, 'Software caused connection abort')
20130307 12:13:18.881 	ERROR 	Calling listener method 'start_keyword' of listener 'C:\Python27\lib\site-packages\robotide\contrib\testrunner\TestRunnerAgent.py' failed: error: (55, 'Software caused connection abort')
20130307 12:13:18.889 	ERROR 	Calling listener method 'end_keyword' of listener 'C:\Python27\lib\site-packages\robotide\contrib\testrunner\TestRunnerAgent.py' failed: error: (55, 'Software caused connection abort')
20130307 12:13:18.897 	ERROR 	Calling listener method 'start_keyword' of listener 'C:\Python27\lib\site-packages\robotide\contrib\testrunner\TestRunnerAgent.py' failed: error: (55, 'Software caused connection abort')
20130307 12:13:18.975 	ERROR 	Calling listener method 'start_keyword' of listener 'C:\Python27\lib\site-packages\robotide\contrib\testrunner\TestRunnerAgent.py' failed: error: (55, 'Software caused connection abort')
20130307 12:13:19.006 	ERROR 	Calling listener method 'end_keyword' of listener 'C:\Python27\lib\site-packages\robotide\contrib\testrunner\TestRunnerAgent.py' failed: error: (55, 'Software caused connection abort')
20130307 12:13:19.022 	ERROR 	Calling listener method 'end_keyword' of listener 'C:\Python27\lib\site-packages\robotide\contrib\testrunner\TestRunnerAgent.py' failed: error: (55, 'Software caused connection abort')
20130307 12:13:19.029 	ERROR 	Calling listener method 'end_test' of listener 'C:\Python27\lib\site-packages\robotide\contrib\testrunner\TestRunnerAgent.py' failed: error: (55, 'Software caused connection abort')
20130307 12:13:19.053 	ERROR 	Calling listener method 'start_test' of listener 'C:\Python27\lib\site-packages\robotide\contrib\testrunner\TestRunnerAgent.py' failed: error: (55, 'Software caused connection abort')
20130307 12:13:19.069 	ERROR 	Calling listener method 'start_keyword' of listener 'C:\Python27\lib\site-packages\robotide\contrib\testrunner\TestRunnerAgent.py' failed: error: (55, 'Software caused connection abort')
20130307 12:13:19.084 	ERROR 	Calling listener method 'end_keyword' of listener 'C:\Python27\lib\site-packages\robotide\contrib\testrunner\TestRunnerAgent.py' failed: error: (55, 'Software caused connection abort')
20130307 12:13:19.092 	ERROR 	Calling listener method 'start_keyword' of listener 'C:\Python27\lib\site-packages\robotide\contrib\testrunner\TestRunnerAgent.py' failed: error: (55, 'Software caused connection abort')
20130307 12:13:19.108 	ERROR 	Calling listener method 'start_keyword' of listener 'C:\Python27\lib\site-packages\robotide\contrib\testrunner\TestRunnerAgent.py' failed: error: (55, 'Software caused connection abort')
20130307 12:13:22.076 	ERROR 	Calling listener method 'log_message' of listener 'C:\Python27\lib\site-packages\robotide\contrib\testrunner\TestRunnerAgent.py' failed: error: (55, 'Software caused connection abort')
20130307 12:13:22.326 	ERROR 	Calling listener method 'log_message' of listener 'C:\Python27\lib\site-packages\robotide\contrib\testrunner\TestRunnerAgent.py' failed: error: (55, 'Software caused connection abort')
20130307 12:13:22.420 	ERROR 	Calling listener method 'log_message' of listener 'C:\Python27\lib\site-packages\robotide\contrib\testrunner\TestRunnerAgent.py' failed: error: (55, 'Software caused connection abort')
20130307 12:13:22.428 	ERROR 	Calling listener method 'end_keyword' of listener 'C:\Python27\lib\site-packages\robotide\contrib\testrunner\TestRunnerAgent.py' failed: error: (55, 'Software caused connection abort')
20130307 12:13:22.451 	ERROR 	Calling listener method 'end_keyword' of listener 'C:\Python27\lib\site-packages\robotide\contrib\testrunner\TestRunnerAgent.py' failed: error: (55, 'Software caused connection abort')
20130307 12:13:22.459 	ERROR 	Calling listener method 'end_test' of listener 'C:\Python27\lib\site-packages\robotide\contrib\testrunner\TestRunnerAgent.py' failed: error: (55, 'Software caused connection abort')
20130307 12:13:22.490 	ERROR 	Calling listener method 'end_suite' of listener 'C:\Python27\lib\site-packages\robotide\contrib\testrunner\TestRunnerAgent.py' failed: error: (55, 'Software caused connection abort')
20130307 12:13:22.498 	ERROR 	Calling listener method 'end_suite' of listener 'C:\Python27\lib\site-packages\robotide\contrib\testrunner\TestRunnerAgent.py' failed: error: (55, 'Software caused connection abort')
20130307 12:13:22.514 	ERROR 	Calling listener method 'end_suite' of listener 'C:\Python27\lib\site-packages\robotide\contrib\testrunner\TestRunnerAgent.py' failed: error: (55, 'Software caused connection abort')

These errors lead to situation where I cannot open RIDE test log / report directly from RIDE 1.1 as the links in the UI are not active.
However I can still access those files by opening manually the folder which RIDE displays at the end of the test execution.

I reinstalled RIDE 1.0.1 over RIDE 1.1 and these errors disappeared
and I can access again RIDE test log &amp; report directly from RIDE links.

I'm using RFW 2.7.6, Python 2.7.3 &amp; Jython 2.5.1

Kaj

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1218#c1) by @mkorpela on 12 Mar 2013*


Could you update Jython to a more recent version?

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1218#c2) by kaj... on 12 Mar 2013*


I'm using Sikuli and this for example doesn't work at all with Jython 2.5.2, that I've tested. Haven't tried with newer Jython versions as I've now a working configuration. I'll try the Jython 2.5.3 / newer when I've time and update here what is the result.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1218#c3) by @mkorpela on 12 Mar 2013*


RIDE doesn't anymore fully support Python / Jython 2.5 series. Thus if you can get your tests running with Jython 2.7beta - I would suggest that you would use that.

If you happen to be working for NSN, please contact core team through internal mailing list - so we can give you better support.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1218#c4) by kaj... on 15 Apr 2013*


Finally had time to experiment the Jython 2.7b1 with RIDE 1.1 and Sikuli &amp; SoapUI.
After I installed the Jython 2.7b1 and tested my SoapUI library with it, that worked immediately and those TestRunnerAgent errors disappeared. But when I tested my Sikuli library, I had an issue with the new Jython as my Sikuli library didn't find the sikuli-script.jar's Sikuli library (from sikuli.Sikuli import *). I tried several different ways to import this library, but not luck with this one. I even rewrote my sikuli library in order to solve this issue, but nothing worked.

Jython 2.5.1 worked like charm with Sikuli. As for Jython 2.7b1 I tried my last trick and extracted the sikuli folder from sikuli-script.jar and placed that folder under jython2.7b1\Lib and after that tried my Sikuli library once again and this time I got 99&#37; of my Sikuli keywords working and no TestRunnerAgent errors appeared. I've to look into those remaining keywords, can I get those working also, so I can use the Jython 2.7b1 with my Sikuli library and RIDE 1.1.

So at this point I confirm that Jython 2.7b1 solves this issue, however there's an issue with Jython 2.7b1 and Sikuli which I couldn't solve the way I expected/wanted and I had to use a trick to circumvent this issue and to get my Sikuli library working.

It should be noted on Wiki/some other place and preferably also on the download site that RIDE 1.1 doesn't anymore fully support Jython/Python 2.5 series. This can be achieved for example by adding SW/system requirements on the Description field which is located under the RIDE download link on RIDE download page, so people could check from there if their machine meets the requirements before installing newer version RIDE.


I used to work for Nokia/NSN for over 12 years, but in 2009 I left the company and accepted new challenges, so the internal mailing list isn't in my disposal anymore.
